### ISSUE TITLE
Supporting initialization options

### ISSUE THREAD

**Issue by fjolliton** [palantir/python-language-server#403]
Hi,

Unless I misread the code, the `init_opts` argument of `Config` is stored but never used. The name `init_opts` appears only in `config.py`.

I'm using PyLS with Emacs, and I was trying to pass initialization options with:

```
(lsp-define-stdio-client
 lsp-python "python"
 lsp-python--get-root '("pyls")
 :extra-init-params '(:pyls (:configurationSources [])))
```

which result in passing:

```
{"pyls": {"configurationSources": []}}
```

to `Config` in the `init_opts` argument. (Here I was trying to disable style linters.)

On the other side, Emacs LSP mode does have an implemantation for `workspace/didChangeConfiguration` but doesn't use it. So the combination of Emacs and PyLS is somewhat complicated.

I'm not sure if the `initializationOptions` argument of the `initialize` LSP function (alas, not documented in the protocol it seems, which doesn't help either) is supposed to receive the same values than with the `workspace/didChangeConfiguration` function, but if so, it would be nice to support it.

Edit: Actually, there are workarounds for Emacs discussed here: https://github.com/emacs-lsp/lsp-mode/issues/167 but if `initializationOptions` is supposed to work the same than `workspace/didChangeConfiguration`. the change would still be welcome.

**Comment by gatesn** [comment]
Yeah I came to the same conclusion as you. It's not obvious that initializationOptions are the same as didChangeConfiguration. At the time, I went with what VSCode sent to the language server. Unless that's changed now I'd be inclined to leave it as-is.
